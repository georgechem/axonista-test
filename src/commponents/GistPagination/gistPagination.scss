@import "../../var.scss";

.gistPagination {
  max-width: $container-width;
  margin: 1em auto;
  background-color: $col-level-1;
  &__bar{
    font: $font-level-1;
    display: flex;
    justify-content: center;
    border: 1px solid $col-level-2;
    &__button {
      text-transform: uppercase;
      padding: $main-spacing;
      cursor: pointer;
      &:hover {
        background-color: $col-level-2;
      }
    }
  }
}

.border {
  border-width: 1px;
  border-style: solid;
}
.left {
  border-color: $col-level-3 transparent $col-level-3 $col-level-3;
}
.center {
  border-color: $col-level-3;
  cursor: none;
}
.right {
  border-color: $col-level-3 $col-level-3 $col-level-3 transparent;
}