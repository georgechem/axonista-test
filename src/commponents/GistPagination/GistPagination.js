import React from 'react';
import PropTypes from 'prop-types';

import './gistPagination.scss';

const GistPagination = ({onNextPageClick, page, onPrevPageClick}) => {

    return (
        <div className="gistPagination">
            <div className="gistPagination__bar">
                <div
                    className="gistPagination__bar__button border left"
                    onClick={onPrevPageClick}
                >Prev</div>
                <div className="gistPagination__bar__button border center">{page}</div>
                <div
                    className="gistPagination__bar__button border right"
                    onClick={onNextPageClick}
                >Next</div>
            </div>
        </div>
    );
};

GistPagination.propTypes = {
    onNextPageClick: PropTypes.func.isRequired,
    page: PropTypes.number.isRequired,
    onPrevPageClick: PropTypes.func.isRequired
}

export default GistPagination;