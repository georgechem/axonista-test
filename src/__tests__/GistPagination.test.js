import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {cleanup, render, screen, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom';
import GistPagination from "../commponents/GistPagination/GistPagination";

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GistPagination
        onNextPageClick={jest.fn()}
        page={1}
        onPrevPageClick={jest.fn()}/>, div);
});

test('calls click on Next and Prev', () => {
    const onNextPageClick = jest.fn();
    const onPrevPageClick = jest.fn();
    render(<GistPagination
        onNextPageClick={onNextPageClick}
        page={1}
        onPrevPageClick={onPrevPageClick}
    />);
    fireEvent.click(screen.getByText(/next/i));
    expect(onNextPageClick).toHaveBeenCalledTimes(1);

    fireEvent.click(screen.getByText(/prev/i));
    expect(onPrevPageClick).toHaveBeenCalledTimes(1);
});

test('matches Snapshot', () => {

    const tree = renderer.create(<GistPagination
        onPrevPageClick={jest.fn()}
        page={1}
        onNextPageClick={jest.fn()}
    />).toJSON();

    expect(tree).toMatchSnapshot();


})