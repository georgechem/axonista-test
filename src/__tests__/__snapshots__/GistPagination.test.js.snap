// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `
<div
  className="gistPagination"
>
  <div
    className="gistPagination__bar"
  >
    <div
      className="gistPagination__bar__button border left"
      onClick={[MockFunction]}
    >
      Prev
    </div>
    <div
      className="gistPagination__bar__button border center"
      data-testid="page"
    >
      1
    </div>
    <div
      className="gistPagination__bar__button border right"
      onClick={[MockFunction]}
    >
      Next
    </div>
  </div>
</div>
`;
